credo issues

lrosa007 opened the pull request:
<strike>Resolved most of the credo issues, except the few (currently 4) in `lib/faker/lorem.ex` bc I wasn't clever enough to find a nice way to do it quickly.</strike> nvm got em all

lrosa007 commented on lib/faker/date.ex:
```diff
@@ -51,9 +51,12 @@ if Version.match?(System.version(), ">= 1.3.0") do
+      |> Kernel.+(unix_now)
       |> DateTime.from_unix!()
       |> DateTime.to_date()
     end
```
I will admit I probably abused the pipes on this one. If other people end up agreeing that the original formatting was better I'll gladly switch it back.

igas replied:
I personally don't like pipes for math, by math I mean not only this example but next few also.
```
Integer.to_string(:crypto.rand_uniform(0, 65_536), 16)
```
This one around numbers and pretty clear for me. We get random number and change base for number (math). But in cases where it's flow of function pipes perfect. If you know what I mean. @tobyhinloopen what do you think?

lrosa007 replied:
I agree. I honestly didn't expect this series of pipes to make it through, it was fun though.

igas commented on lib/faker/internet.ex:
```diff
@@ -113,7 +111,9 @@ defmodule Faker.Internet do
   @spec ip_v6_address() :: String.t
   def ip_v6_address do
     Enum.map_join 1..8, ":", fn(_part) ->
-      Integer.to_string(:crypto.rand_uniform(0, 65_536), 16)
```
I would keep this one. See other comment.

lrosa007 replied:
I did something slightly different, let me know if this is ok

igas commented on lib/faker/internet.ex:
```diff
@@ -123,10 +123,16 @@ defmodule Faker.Internet do
   @spec mac_address() :: String.t
   def mac_address do
-    Enum.map_join(1..6, ":", fn(_part) ->
-      Integer.to_string(:crypto.rand_uniform(0, 256), 16)
```
and this one.

lrosa007 replied:
see comment above, same thing was done here

igas commented:
lgtm except comments. thanks for contribution.

lrosa007 commented:
I adjusted the stuff you commented on. To still avoid starting a pipe chain without a raw value, I took the math stuff and binded the result to a variable and then started the pipe chain with that.

I also rebased and some new credo issues came up so I took care of those as well. Mostly the standard `pipe chain must start with raw value` kind of issue.

lrosa007 commented:
ping @igas :)

tobyhinloopen commented on USAGE.md:
```diff
@@ -177,6 +179,44 @@ Faker.Date.date_of_birth(10..19) #=> ~D[2004-05-15]
+to = DateTime.utc_now() |> DateTime.to_unix
+to = to + 86400 * 10 |> DateTime.from_unix!
+
+Faker.DateTime.between(from, to)
```
I really would like this to be inlined. I'd suggest using `~N[]` to create literal naive datetimes, since the function is compatible with naive datetime types

tobyhinloopen commented:
LGTM, except for the one comment on `USAGE.md`

lrosa007 commented:
@tobyhinloopen ok did it. Let me know if that is not what you meant.

igas commented:
@lrosa007 hey, can you pls rebase, you should not have @anthonator commit in it.

lrosa007 commented:
@igas yea no problem, I think I fudged up a rebase earlier last week

lrosa007 commented on USAGE.md:
```diff
@@ -177,6 +179,40 @@ Faker.Date.date_of_birth(10..19) #=> ~D[2004-05-15]
+#=>  std_offset: 0, time_zone: "Etc/UTC", utc_offset: 0, year: 2016,
+#=>  zone_abbr: "UTC"}
+
+Faker.DateTime.between(~N[2016-12-20 00:00:00], ~N[2016-12-25 00:00:00])
```
although, having his commit here allowed me to adjust this I believe

lrosa007 commented:
@igas ok I fixed the ugly rebase I did earlier and I no longer have @anthonator commit in this branch. Also I fixed the `lorem.ex` credo issues; it was obvious once I went back to looked at them again. So now every credo issue is resolved not just most.

igas commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
